### PR TITLE
Implement scrollbar-width CSS parsing

### PR DESCRIPTION
**Pull request by lukewarlow**
#### 98d2001782e57b75e18d1cccb1beb4a8db83b242
<pre>
Implement scrollbar-width CSS parsing
<a href="https://bugs.webkit.org/show_bug.cgi?id=256796">https://bugs.webkit.org/show_bug.cgi?id=256796</a>

Reviewed by Tim Nguyen.

Spec: <a href="https://drafts.csswg.org/css-scrollbars/#scrollbar-width">https://drafts.csswg.org/css-scrollbars/#scrollbar-width</a>

Grammar for property: auto | thin | none

This adds the property to the CSS parser behind the flag, and updates the relevant expectation files

* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/inheritance-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-scrollbars/scrollbar-width-parsing-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-typed-om/the-stylepropertymap/properties/scrollbar-width-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/property-list.js:
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/ipad/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/css/css-cascade/all-prop-revert-layer-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/accumulation-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/addition-per-property-002-expected.txt:
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/web-animations/animation-model/animation-types/interpolation-per-property-002-expected.txt:
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/css/css-cascade/all-prop-initial-xml-expected.txt:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
(WebCore::CSSPropertyAnimationWrapperMap::CSSPropertyAnimationWrapperMap):
* Source/WebCore/css/CSSPrimitiveValueMappings.h:
* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle):
* Source/WebCore/css/parser/CSSPropertyParser.cpp:
(WebCore::initialValueForLonghand):
* Source/WebCore/rendering/style/RenderStyle.cpp:
(WebCore::RenderStyle::initialScrollbarWidth):
(WebCore::RenderStyle::scrollbarWidth const):
(WebCore::RenderStyle::setScrollbarWidth):
* Source/WebCore/rendering/style/RenderStyle.h:
* Source/WebCore/rendering/style/RenderStyleConstants.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/rendering/style/RenderStyleConstants.h:
* Source/WebCore/rendering/style/StyleRareNonInheritedData.cpp:
(WebCore::StyleRareNonInheritedData::StyleRareNonInheritedData):
(WebCore::StyleRareNonInheritedData::operator== const):
* Source/WebCore/rendering/style/StyleRareNonInheritedData.h:

Canonical link: <a href="https://commits.webkit.org/264214@main">https://commits.webkit.org/264214@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/e1411c8b6cf71854061155200d776cbf2004db4c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7002 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/7242 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/7422 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/8614 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7236 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/7010 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/8505 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7177 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10150 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7127 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/7798 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/6461 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8717 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/5166 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/6369 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/14149 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/6835 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/6463 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/9320 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6937 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/5685 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/6306 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1666 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/10499 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6688 "Built successfully") | | |

### CODE REVIEW

**Comment by Ahmad-S792** [comment]
@lukewarlow - I know it is draft but can you add 'spec' link for reference in commit message. Thanks!

File: Source/WebCore/css/ComputedStyleExtractor.cpp
Comment (nt1m) [inline Source/WebCore/css/ComputedStyleExtractor.cpp]: I think this can just be `return createConvertingToCSSValueID(style.scrollbarWidth());`

`DEFINE_TO_FROM_CSS_VALUE_ID_FUNCTIONS` makes this work.

**Review by nt1m — APPROVED** [review]

**Review by nt1m — APPROVED** [review]

**Comment by lukewarlow** [comment]
All the pipeline appear to be passing so idk why it says 1 of the checks failed?

image

**Comment by Ahmad-S792** [comment]
> All the pipeline appear to be passing so idk why it says 1 of the checks failed?
> 
> 
> 
> image
> 
> 

I would suggest posting link to this in help channel on Slack Channel and tagging @aj062 & @JonWBedard and they look into EWS so might know whether it is bug to ignore or something needs to be fixed. Meanwhile if you want, I can commit it to 'main' for you.

**Comment by lukewarlow** [comment]
I've posted that. If you can merge it that'd be great but no worries if you'd rather wait to see if there is a pipeline issue

**Comment by lukewarlow** [comment]
I see graouts is a codeowner (does this mean he has to approve?) but he's on leave. Fwiw he reviewed and approved the relevant code in #10285

**Comment by Ahmad-S792** [comment]
> I see graouts is a codeowner (does this mean he has to approve?) but he's on leave. Fwiw he reviewed and approved the relevant code in #10285 

I think 'codeowner' is not followed in Webkit as much. So Tim's review is fine. Will add 'merge-queue' now.

**Comment by JonWBedard** [comment]
For folks looking at this PR in the future:
Looks like GitHub dropped one of our queue's update to the commit status when we re-tried the build (in any event, the failed build was a pre-existing test failure on Debug that was infrequent enough to slip bellow our automated bar for pre existing test failures)

As for "code owners", WebKit is using that feature to replace our watchlist. And we would rename the file if we could. Basically, "code owners" are individuals who would like to be notified when changes to certain files are proposed, but a review from a "code owner" is not generally required. The WebKit project expects that reviewers only review code they are qualified to review, that burden falls on the review not the code author.

**Comment by lukewarlow** [comment]
I did not. Would that be the animationstests?

**Comment by graouts** [comment]
That would be anything under `LayoutTests/imported/w3c/web-platform-tests/` that isn't an `-expected.txt` file. There is a script `Tools/Scripts/export-w3c-test-changes` to help you with this, but it's not always easy to use and I'm not sure how to use it after your branch landed. Maybe @nt1m or @gsnedders can provide pointers here. Best option might be to make a manual PR against http://github.com/web-platform-tests/wpt/.

**Comment by lukewarlow** [comment]
Fwiw all the tests inside of /css/css-scrollbars was imported from wpt. So i think it might just be the animation property-list but thinking about it I didn't actually check if that needed importing so it might be it's already handled in WPT.

**Comment by lukewarlow** [comment]
@graouts I've made a manual PR https://github.com/web-platform-tests/wpt/pull/40124

**Pull request merged**
